plugin/client_rpc.go

```diff
@@ -395,26 +395,31 @@ func (g *hooksRPCClient) FileWillBeUploaded(c *Context, info *model.FileInfo, fi
 		serveIOReader(file, uploadedFileConnection)
```

Comment:
@lieut-data 
should the func also block till this operation is complete as well?

Reply:
No, there's no specific reason to explicitly coordinate with that goroutine -- the second goroutine won't finish until the reader closes automatically.

Reply:
@lieut-data 
And i guess `g.client.Call("Plugin.FileWillBeUploaded", _args, _returns); ` is safe to use `uploadedFileStreamId` even if this go-routine hasn't finished writing to `uploadedFileConnection`?

Reply:
just curious if `g.client.Call("Plugin.FileWillBeUploaded", _args, _returns);` is safe to call with the respective stream-ids before both these go-routines are finished?

Reply:
Yes, that's correct: in fact, no data should be copied at all unless the plugin reads from the `io.Reader`.